data/src/main/java/com/netflix/iceberg/data/TableScanIterable.java

```diff
@@ -125,7 +125,10 @@ public void close() throws IOException {
     private Iterator<Record> currentIterator = emptyIterator();
 
     private ScanIterator(Iterable<CombinedScanTask> tasks) {
-      this.tasks = Lists.newArrayList(concat(transform(tasks, CombinedScanTask::files))).iterator();
+      this.tasks = StreamSupport.stream(tasks.spliterator(), false)
```

Comment:
I think this is an unrelated change.

Reply:
Oh weird, will revert but made this change because my IDE was complaining and refuses to compile, but `./gradlew . check` works fine.